src/context.ts

```diff
@@ -42,13 +42,9 @@ function createContext<S extends StoreApi<unknown>>() {
   }) => {
     const storeRef = useRef<S>()
 
-    if (!storeRef.current) {
-      storeRef.current = createStore()
-    }
-
     return createElement(
       ZustandContext.Provider,
-      { value: storeRef.current },
+      { value: (storeRef.current ||= createStore()) },
```

Comment:
I find such shorthand less readable than an explicit `if`, which purposefully takes some space to highlight a certain condition.

Also, nullish coalescing assignment would probably make more sense here:

```suggestion
      { value: (storeRef.current ??= createStore()) },
```

But again, I disagree with this change. I don't want to do bike shedding here though, so I won't block it and leave the decision-making to @dai-shi :)

Reply:
I actually like `||=` and I would use it when it makes sense.
For this PR, it's not important as we will be dropping it in the future version.